.github/workflows/test.yml

```diff
@@ -0,0 +1,160 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
```suggestion
```

For trouble shooting purpose, we may do the kernel-related testing in the PR level to ensure we fix the incompatible issue before we merge it to main branch.